pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java

```diff
@@ -173,6 +188,25 @@ public byte[] toBytes()
       dataOutputStream.writeInt(bytes.length);
       dataOutputStream.write(bytes);
     }
+
+    // Write colum field specs.
+    // todo: consider serializing only defaultNullValues since this is the only required info.
+    int fieldSpecsSize = _columnFieldSpecs == null ? 0 : _columnFieldSpecs.length;
+    dataOutputStream.writeInt(fieldSpecsSize);
```

Comment:
(MAJOR) This is backward incompatible change. The new broker won't understand the data schema returned from the old servers

Reply:
Will take care of ensuring backward compatibility in final PR .. this is just a draft PR to get the discussion going.